showdown/battle.py

```diff
@@ -622,3 +623,36 @@ def __eq__(self, other):
 
     def __repr__(self):
         return "{}".format(self.name)
+
+
+class PersistentCondition:
+
+    def __init__(self):
+        self.name = ""
+        self.setter = None
+        self.elapsed_turns = -1
+
+    def to_dict(self):
+        return {
+            "name": self.name,
+            "setter": self.setter.to_dict(),
+            "elapsed_turns": self.elapsed_turns,
+        }
+
+    def get_remaining_turns(self):
+        if self.name == constants.TRICK_ROOM:
+            return 4-self.elapsed_turns-1
+        if self.setter == None:
+            return 5-self.elapsed_turns-1
+
+        total_turns = 5
+        if (
+                self.name == constants.RAIN and self.setter.item == "damprock" or
+                self.name == constants.SUN and self.setter.item == "heatrock" or
+                self.name == constants.HAIL and self.setter.item == "icyrock" or
+                self.name == constants.SAND and self.setter.item == "smoothrock" or
+                self.name in constants.TERRAINS and self.setter.item == "terrainextender"
+        ):
+            total_turns+=3
```

Comment:
This only checks the pokemon's item at the time of the call. The item could have been knocked off / tricked / etc. by the time this is called.